sdk/apimanagement/api-management-custom-widgets-tools/src/node/deploy.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { CONFIG_FILE_NAME } from "../paths";
+import CustomWidgetBlobService from "./CustomWidgetBlobService";
+import fs from "fs";
+import getStorageSasUrl from "./getStorageSasUrl";
+import readdir from "./readdir";
+
+/**
+ * resourceId - resource ID of API Management service "subscriptions/[subscription-id]/resourceGroups/[resource-group-name]/providers/Microsoft.ApiManagement/service/[service-name]"
+ * managementApiEndpoint - URL with protocol (e.g. https://management.azure.com)
+ * apiVersion - optional to override default (e.g. "2019-01-01")
+ * tokenOverride - optional, provides token to use for auth, instead of 'az login' approach
+ */
+export type TServiceInformation = {
+  resourceId: string;
+  managementApiEndpoint: string;
+  apiVersion?: string;
+  tokenOverride?: string;
+};
+
+/**
+ * Deploys everything from /dist folder to the API Management DevPortals' blob storage.
+ *
+ * @param serviceInformation - service information for deployment
+ * @param name - name of the widget to be deployed
+ * @param fallbackConfigPath - local path to the config file (by default "./static/config.msapim.json")
+ */
+async function deploy(
+  serviceInformation: TServiceInformation,
+  name: string,
+  fallbackConfigPath = "./static/" + CONFIG_FILE_NAME
+): Promise<void> {
+  console.log("\n\n");
+  console.log("Starting deploy process of custom widget: " + name);
+  console.log("Please, sign in to your Azure account when prompted\n");
```

Comment:
We need to remove all console.logs before merging. If you need logging please use @azure/logger. Look at the template project for guidance

Reply:
So it should look something like this, is that correct?
```
const logger = createClientLogger("deploy");
logger.info("\n\n");
logger.info("Starting deploy process of custom widget: " + name);
logger.info("Please, sign in to your Azure account when prompted\n");
```

Reply:
no, azure loger logs to console and allows setting verbosity level through Environment Variables

Reply:
You can also set logging verbosity programatically